index.bs

```diff
@@ -1649,54 +1649,69 @@ session to exist at any particular time, or by providing more complicated sessio
-1. Check if at least one of the specified combinations of {{PublicKeyCredentialType}} and cryptographic parameters is supported.
+1. Check if at least one of the specified combinations of {{PublicKeyCredentialType}} and cryptographic parameters in
+    |credTypesAndPubKeyAlgs| is supported.
     If not, return an error code equivalent to "{{NotSupportedError}}" and terminate the operation.
-1. Check if a credential matching any of the supplied {{PublicKeyCredential}} identifiers is present on this authenticator. If
+1. Check if a credential matching an item of |excludeCredentialDescriptorList| is present on this authenticator. If
     so, return an error code equivalent to "{{NotAllowedError}}" and terminate the operation.
 1. If |requireResidentKey| is |true| and the authenticator cannot store a [=Client-side-resident Credential
     Private Key=], return an error code equivalent to "{{ConstraintError}}" and terminate the operation.
 1. If |requireUserVerification| is |true| and the authenticator cannot perform user verification,
     return an error code equivalent to "{{ConstraintError}}" and terminate the operation.
 1. Prompt the user for consent to create a new credential. The prompt for obtaining this consent is shown by the authenticator
-    if it has its own output capability, or by the user agent otherwise. If the user denies consent, return an error code
+    if it has its own output capability, or by the user agent otherwise. This prompt can use information in |rpEntity| and
+    |userEntity|. If the user denies consent, return an error code
     equivalent to "{{NotAllowedError}}" and terminate the operation.
 1. Once user consent has been obtained, generate a new credential object:
-    1. Generate a set of cryptographic keys using the most preferred combination of {{PublicKeyCredentialType}} and cryptographic
-        parameters supported by this authenticator.
-    1. Generate an identifier for this credential, such that this identifier is globally unique with high probability across all
+    1. Let |publicKey|/|privateKey| be a new set of cryptographic keys using the combination of {{PublicKeyCredentialType}} and
+        cryptographic parameters represented by the first item in |credTypesAndPubKeyAlgs| that is supported by this authenticator.
+    1. Let |credentialId| be a new identifier for this credential that is globally unique with high probability across all
         credentials with the same type across all authenticators.
-    1. Associate the credential with the specified [=RP ID=] and the user's account identifier
-        {{MakePublicKeyCredentialOptions/user}}.{{PublicKeyCredentialUserEntity/id}}.
-    1. Delete any older credentials with the same [=RP ID=] and {{MakePublicKeyCredentialOptions/user}}.{{PublicKeyCredentialUserEntity/id}}
-        that are stored locally by the [=authenticator=].
+    1. Associate the |credentialId| and |privateKey| with |rpId| and the user's account identifier
```

Comment:
We could model this association as a map, which would be clear about when a new credential has to overwrite an old one. Do we want to?

Reply:
wrt modeling as map: i dunno, that might be interpreted as overly-prescriptive (by platform- & authnr-type folk).

wrt store more than the credential ID and private key:  is this not saying to store (at least) the 4-tuple:  ( |credentialID|, |privateKey|, |rpId|, |userEntity|.{{PublicKeyCredentialUserEntity/id}} ) ?

Reply:
Yeah, which isn't enough to return the User Handle from https://w3c.github.io/webauthn/#op-get-assertion as demanded by #558.

Reply:
'k, picking one platform and one authenticator person arbitrarily: @leshi and @rlin1, am I going to break something terribly here if I say that the authenticator maintains a [map](https://infra.spec.whatwg.org/#maps) from (rpId, userEntity.id) to (credentialId, privateKey), which we'll then look things up from within getAssertion?

Reply:
adding a person from yet another platform: @akshayku

Reply:
seems to me the concern here ( as expressed by platform/browser folk in past) is "overly constrained/specified", rather than "break"  :)

Reply:
After trying to model this as a map, it turns out that the map doesn't cover non-resident keys, and modeling those more precisely takes some work. So never mind about this question.